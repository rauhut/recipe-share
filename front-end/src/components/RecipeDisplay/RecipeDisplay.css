.recipe-card-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(340px, 1fr));
  grid-gap: 2%;
  padding: 20px;
  align-items: center;
  justify-content: center;
}

.card-title {
  font-weight: bolder;
}
