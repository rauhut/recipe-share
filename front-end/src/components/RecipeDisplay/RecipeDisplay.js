import React from "react";
import {
  Card,
  CardImg,
  CardTitle,
  CardText,
  CardSubtitle,
  CardBody,
} from "reactstrap";
import { Link } from "react-router-dom";

const RecipeDisplay = ({ recipes }) => {
  return (
    <div className="recipe-card-grid">
      {recipes.map((recipe) => {
        return (
          <Card>
            <CardImg
              top
              width="100%"
              src={recipe.picture}
              alt="Card image cap"
            />
            <CardBody>
              <CardTitle>{recipe.name}</CardTitle>
              <CardSubtitle>{recipe.cookTime}</CardSubtitle>
              <CardText>{recipe.description}</CardText>
              <Link to={`/recipe/${recipe._id}`}>
                <button className="button-secondary">View Recipe</button>
              </Link>
            </CardBody>
          </Card>
        );
      })}
    </div>
  );
};

export default RecipeDisplay;
