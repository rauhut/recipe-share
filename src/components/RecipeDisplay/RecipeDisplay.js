import React from "react";
import {
  Card,
  CardImg,
  CardTitle,
  CardText,
  CardSubtitle,
  CardBody,
} from "reactstrap";
import { Link } from "react-router-dom";
import "./RecipeDisplay.css";
import defaultImage from "../../Default.jpg";

const RecipeDisplay = ({ recipes }) => {
  return (
    <div className="recipe-card-grid">
      {recipes.map((recipe) => {
        return (
          <Card key={`${recipe._id}`}>
            <CardImg
              top
              width="100%"
              src={recipe.picture}
              onError={(e) => {
                e.target.onerror = null;
                e.target.src = defaultImage;
              }}
              alt={`${recipe.name}`}
            />
            <CardBody>
              <CardTitle>{recipe.name}</CardTitle>
              <CardSubtitle>{recipe.cookTime}</CardSubtitle>
              <CardText>{recipe.description}</CardText>
              <Link to={`/recipe/${recipe._id}`}>
                <button className="button-secondary">View Recipe</button>
              </Link>
            </CardBody>
          </Card>
        );
      })}
    </div>
  );
};

export default RecipeDisplay;
