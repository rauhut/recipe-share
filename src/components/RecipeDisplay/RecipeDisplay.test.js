import React from "react";
import { shallow } from "enzyme";
import RecipeDisplay from "./RecipeDisplay";

const recipes = [
  {
    name: "cookies",
    ingredients: ["flour", "sugar", "butter", "salt", "chocolate"],
    steps: ["step 1", "step 2", "step 3"],
    cookTime: "30 mins",
    description: "just basic cookies",
  },
  {
    name: "meatballs",
    ingredients: ["ground beef", "eggs", "bread crumbs", "garlic", "onion"],
    steps: ["step 1", "step 2", "step 3"],
    cookTime: "30 mins",
    description: "just basic meatballs",
  },
];

describe("RecipeDisplay", () => {
  it("renders without crashing", () => {
    expect(
      shallow(<RecipeDisplay recipes={recipes} />).debug()
    ).toMatchSnapshot();
  });
});
