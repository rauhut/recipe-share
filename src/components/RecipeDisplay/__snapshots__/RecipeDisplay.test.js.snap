// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDisplay renders without crashing 1`] = `
"<div className=\\"recipe-card-grid\\">
  <Card tag=\\"div\\">
    <CardImg top={true} width=\\"100%\\" src={[undefined]} alt=\\"Card image cap\\" tag=\\"img\\" />
    <CardBody tag=\\"div\\">
      <CardTitle tag=\\"div\\">
        cookies
      </CardTitle>
      <CardSubtitle tag=\\"div\\">
        30 mins
      </CardSubtitle>
      <CardText tag=\\"p\\">
        just basic cookies
      </CardText>
      <Link to=\\"/recipe/undefined\\">
        <button className=\\"button-secondary\\">
          View Recipe
        </button>
      </Link>
    </CardBody>
  </Card>
  <Card tag=\\"div\\">
    <CardImg top={true} width=\\"100%\\" src={[undefined]} alt=\\"Card image cap\\" tag=\\"img\\" />
    <CardBody tag=\\"div\\">
      <CardTitle tag=\\"div\\">
        meatballs
      </CardTitle>
      <CardSubtitle tag=\\"div\\">
        30 mins
      </CardSubtitle>
      <CardText tag=\\"p\\">
        just basic meatballs
      </CardText>
      <Link to=\\"/recipe/undefined\\">
        <button className=\\"button-secondary\\">
          View Recipe
        </button>
      </Link>
    </CardBody>
  </Card>
</div>"
`;
